### ISSUE TITLE
bug: Select Element not working on Windows 10

### ISSUE THREAD

**Issue by pjc2007** [driftyco/ionic#4803]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>desktop</span>  <span ionic-webview>browser</span>

<span ionic-description>Hi, this is an issue on Windows 10 Tablet (Mobile), not desktop; I put desktop as cannot see a platform for Windows 10 yet.

Anyway, I have downloaded and linked in version Ionic 1.2, but I still can't get the Select element to work; they do not open. Also the text alignment seems to be aligned to the top, can even see this on the desktop edge browser.

I have used the following markup for version 1.2

```html
<ion-input class="item item-input item-select">
  <i class="icon ion-person"></i>
    <ion-label>please select</ion-label>
      <select>
        <option>Option1</option>
        <option selected>Option 2</option>
        <option>Option 3</option>
    </select>
</ion-input>
```

This was an issue in the previous version, but unfortunately still seems to be there in 1.2. I've tried playing with the css but haven't been able to get it working. 

Cheers</span>

<span is-issue-template></span>

**Comment by PerfectionCSGO** [comment]
Notably, the select element opens itself a second time after selecting an option on Windows Phone 8.1. Or sometimes requires more than one tap to open up.

**Comment by pjc2007** [comment]
I gave this a try but I couldn't get it to open on a second tap, but if I did continuously tap ,sometimes after 5 to 10 it would open.... but then you couldn't select anything. But only ever now and then I could get this to work at all. (on the Windows 10 tablet).

**Comment by PerfectionCSGO** [comment]
If you're still looking for a solution, this is mine that works in WP8.1 phone and W10 emulator.

angular.module('starter').directive('select', selectDirective);
selectDirective.$inject = [];
function selectDirective() {
    return {
        restrict: 'E',
        replace: false,
        link: function (scope, element) {
            if (ionic.Platform && ionic.Platform.isWindowsPhone()) {
                element.attr('data-tap-disabled', 'true');
            }
        }
    };
}

Simply put, for windows phone you want to set data-tap-disabled=true on select directives.

**Comment by pjc2007** [comment]
@PerfectionCSGO Thankyou enormously for this!! So, on the Windows 10 table, the platform comes in as **'edge'**, so using your code above, I just need to replace the platform test with the line..
    
    if (ionic.Platform && ionic.Platform.platform() === "edge") {

Now that this worked,  looking at every property of the select in Edge vs Chrome, I found alignment problem to just be the height... In Chrome the select height is `45px`, where as Edge had it as `22px`. So together with your fix, also adding.

    select {
      height:45px
    }

the select now appear to work on the Windows tablet exactly as it does on Android. 

This is a great workaround.

**Comment by jdnichollsc** [comment]
+1

**Comment by PerfectionCSGO** [comment]
@jdnichollsc I'd argue while you have a nice solution there, mine is a bit easier to implement and is more consistent with showing users a familiar input method.

**Comment by jdnichollsc** [comment]
@PerfectionCSGO I'm agree with you, but your solution doesn't work in Windows 10 Remote Machine :/

**Comment by JonHerr** [comment]
I have the same issue with windowsPhone10 and windows 10. data-tap-disable = true fixes this. not sure if adding to the element for all iOS and Android would be fine. but using the directive works.

**Comment by chazwoza** [comment]
Can someone provide a full code example here?

**Comment by JonHerr** [comment]
in app.js after you declare
    
    myApp = angular.module( 'app', []);

add this directive

    myApp.directive('select', selectDirective);
    selectDirective.$inject = [];
    function selectDirective() {
        return {
            restrict: 'E',
            replace: false,
            link: function (scope, element) {
                if (ionic.Platform && ionic.Platform.isWindowsPhone() || ionic.Platform.isEdge()) {
                    element.attr('data-tap-disabled', 'true');
                }
            }
        };
    }

you will want to add a platform specific css to your sass file or in /www/css/style.css with

    select {
      height:45px
    }

**Comment by chazwoza** [comment]
hmm - version of ionic doesn't seem to have isEdge method

**Comment by JonHerr** [comment]
well it's in ionic version 1.2.4 not sure when it was added

**Comment by jdnichollsc** [comment]
See my example: https://github.com/jdnichollsc/Ionic-Starter-Template/blob/master/app/js/issues.js

**Comment by chazwoza** [comment]
Doesn't work I'm afraid :(

**Comment by jdnichollsc** [comment]
@chazwoza include my issues.js file in your project, and test again ;)

**Comment by chazwoza** [comment]
Hi @jdnichollsc  - Copied your code as-is into my app.js and still wont work: http://pastebin.com/09iuadyY

Am i including it correctly?

**Comment by chazwoza** [comment]
Any help? :)

**Comment by maqjav** [comment]
In a tablet with Windows 8.1 PRO you need to use the next code:

     if (ionic.Platform && ionic.Platform.platforms.indexOf("win32")) {
          element.attr('data-tap-disabled', 'true');`
     }

**Comment by jgw96** [comment]
Hello all! Thanks for opening an issue with us. I am going to close this one as we don't actually officially support Windows 10 with Ionic 1(although we do with Ionic 2). Thanks for using Ionic everyone!

**Comment by mobidev111** [comment]
please re-open - I understand that you support windows 10 since 1.2: 

"Windows 10 - Ionic 1.2 marks the first release with a new supported platform. Today, Ionic 1.2 supports Windows 10 and Edge, which runs on the new Windows Phone 10 platform." http://blog.ionic.io/announcing-ionic-1-2/

**Comment by jdnichollsc** [comment]
+1

**Comment by ZakkGriffiths** [comment]
Hi guys. Here's an alternative I've just found that doesn't need a platform specific CSS class:

`<.item-select select {
    height: 100%;
}>`

It must be used in combination with the 'select' directive from this thread (or similarly named).
It works in Chrome, Native Windows 10 and on a Windows 10 Lumia Phone, where they all behave similarly. It has the bonus of making the touch-responsive area the full height of the parent ion-list-item(?), rather than requiring users to touch the default option text itself (a much smaller target).

**Comment by chazwoza** [comment]
Thanks @ZakkGriffiths 

I actually gave up on the select and solved my use case by creating my own modal which displayed an item list then called a custom "select" function when I clicked on the item.

**Comment by tundak** [comment]
For windows 10 device. ionic select option crash my app with error:
```
'WWAHost.exe' (Script): Loaded 'Script Code (MSAppHost/2.0)'. 
The program '[8844] WWAHost.exe' has exited with code -1073741819 (0xc0000005) 'Access violation'.
```

any help for this ?
